syntax = "proto3";

package main;

service TrafficRegulation {
  rpc ReceiveDataForLogs (TrafficDataForLogs) returns (TrafficDataForLogsReceiveResponse);
  rpc GetTodayControlLogs (IntersectionRequestForLogs) returns (TrafficRegulationResponse);
  rpc GetLastWeekControlLogs (IntersectionRequestForLogs) returns (TrafficRegulationResponse);
}

message TrafficDataForLogs {
  int32 intersection_id = 1;
  int32 signal_status_1 = 2;
  int32 vehicle_count = 3;
  bool incident = 4;
  string date = 5;
  string time = 6;
}

message TrafficDataForLogsReceiveResponse {
  string message = 1;
}

message IntersectionRequestForLogs {
  int32 intersection_id = 1;
}

message TrafficRegulationResponse {
  int32 intersection_id = 1;
  repeated string logs = 2;
}
